ddtrace/provider.py

```diff
@@ -79,13 +79,36 @@ def __call__(self, *args, **kwargs):
         """
         return self.active()
 
+    def _executor_id(self):
+        # type: () -> int
+        return 0
```

Comment:
True. Threads weren't considered much here because we don't have autoinstrumentation for it and recommend manually doing the span->context propagation. We can easily add thread support here by using thread ids so good call :+1:

Reply:
will it cause any issues? like what happens during a fork?

Reply:
Threads aren't inherited/copied when a fork occurs so I don't think it's an issue. It will be worth seeing what happens with the main thread id however (if it changes or not). If it does actually change in the new process then we might have more generally solved the logic that is done here: https://github.com/DataDog/dd-trace-py/blob/508f991af9f348b8fcd6453dc47a2bb4e8a19a21/ddtrace/tracer.py#L467-L486

Reply:
yeah, that was my concern, the main thread id.

Reply:
```python
import os, threading
print(threading.get_ident())
os.fork()
print(threading.get_ident())
```

```
~/d/dd-trace-py ❮❮❮ python test.py
4602654208
4602654208
4602654208
```

Reply:
I suppose not too surprised there, means it is effectively the same as defaulting to `0`, but could help us in the case that a span/context is passed to another thread?

Reply:
> with the main thread id however (if it changes or not)

you can't really assume anything here IMHO for portability.

Reply:
> you can't really assume anything here IMHO for portability.

I still can't decide if that means we are better off using `threading.get_ident()` or not, because if we do fork and the thread id changes, should we continue to keep the forked spans "active"?

Reply:
We already handle the forking case with https://github.com/DataDog/dd-trace-py/blob/508f991af9f348b8fcd6453dc47a2bb4e8a19a21/ddtrace/tracer.py#L467-L486

Threads should be handled by manually doing the span->context propagation as outlined in our docs.

So I don't think it's necessary to do anything here.